### PR TITLE
Cli 6 rc.0

### PR DESCRIPTION
**Pull request by filipesilva**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
Angular CLI 1.7.3 is used.

## What is the new behavior?
Angular CLI 6.0.0-rc.0 is used.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: aio/angular.json [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+            ],
+            "scripts": []
+          },
+          "configurations": {
+            "next": {
+              "optimization": true,
+              "outputHashing": "all",
+              "sourceMap": false,
+              "extractCss": true,
+              "namedChunks": false,
```
Comment (gkalpak) [inline aio/angular.json]: Why unnamed chunks? I thought we were using named before.

File: aio/angular.json [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+                "input": "src/styles.scss"
+              }
+            ],
+            "scripts": []
+          },
+          "configurations": {
+            "next": {
+              "optimization": true,
+              "outputHashing": "all",
+              "sourceMap": false,
```
Comment (gkalpak) [inline aio/angular.json]: Why no sourcemaps?

File: aio/angular.json [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+    "@schematics/angular:component": {
+      "inlineStyle": true,
+      "prefix": "aio",
+      "styleext": "scss"
+    },
+    "@schematics/angular:directive": {
+      "prefix": "aio"
+    }
+  }
+}
```
Comment (gkalpak) [inline aio/angular.json]: Wow! What a verbose file :grin:

**Review by gkalpak — COMMENTED** [review]
I guess eventually we also need to update this line:

https://github.com/filipesilva/angular/blob/eb852e5a213113821d40c7ec1200ccd35daa1d3b/aio/scripts/deploy-to-firebase.sh#L90

File: aio/src/tsconfig.spec.json [hunk 4]
```diff
@@ -11,11 +11,12 @@
     ]
   },
   "files": [
-    "test.ts"
+    "test.ts",
+    "polyfills.ts"
```
[the remark this answers is not in the document]
Comment (filipesilva) [reply aio/src/tsconfig.spec.json]: Finally fixing an old workaround from CLI 1.0. Unit tests use polyfills but were not including them. We had to force-include that file in the TS compilation. That lead to weird situations.

File: aio/package.json [hunk 5]
```diff
@@ -21,7 +21,7 @@
     "test": "yarn check-env && ng test",
     "pree2e": "yarn check-env && yarn update-webdriver",
     "e2e": "ng e2e --no-webdriver-update",
-    "e2e-prod": "yarn e2e --environment=dev --target=production",
+    "e2e-prod": "yarn e2e --prod",
```
[the remark this answers is not in the document]
Comment (filipesilva) [reply aio/package.json]: Now the CLI works on the basis of "configurations", that modify the existing options. Before there was only really the `production` configuration that was activated via the `--prod` flag, but you couldn't control what it changes.

So now I added the `production` configuration on the `angular.json` file with the options that `--target=production` had before, and used it here.

File: aio/angular.json (hunk 1 above)
Comment (filipesilva) [reply aio/angular.json]: `target=production` always set named chunks to false.

File: aio/angular.json (hunk 2 above)
Comment (filipesilva) [reply aio/angular.json]: `target=production` always set sourcemaps to false.

File: aio/angular.json (hunk 3 above)
Comment (filipesilva) [reply aio/angular.json]: Yes. We might transfer more options to defaults but tbh for anything that has a path+filename I don't think we should have defaults.

File: aio/angular.json (hunk 2 above)
Comment (gkalpak) [reply aio/angular.json]: But why have sourcemaps in `stable` config and not `next` or `archive`. These three should be mostly the same.

File: aio/angular.json (hunk 1 above)
Comment (gkalpak) [reply aio/angular.json]: I am 99% sure we used named chunks before.
<sub>(I am 100% sure we used named chunks before, but I am 99% sure I understand what named chunks are :grin:)</sub>

File: aio/package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (gkalpak) [reply aio/package.json]: Yeah, I reviewed the files in reverse order :grin:
So, do I understand correctly that `--prod` is an alias for `--configuration=production`, which (based on the current config) has the same options as `--target=production` but uses the default ("dev") `environment.ts` file?

[the remark this answers is not in the document]
Comment (filipesilva) [reply aio/package.json]: Yes. I'm not too familiar with the environments/targets used in AIO and why they are used so these options will probably be adapted to suit your needs better in the future.

File: aio/angular.json (hunk 1 above)
Comment (filipesilva) [reply aio/angular.json]: Was it set in options somewhere? I didn't see flags for it in the package.json scripts.

File: aio/angular.json (hunk 2 above)
Comment (filipesilva) [reply aio/angular.json]: Up to you really. The upgrade script tried to infer these bug it cannot infer anything that isn't set down in configs.

File: aio/angular.json (hunk 1 above)
Comment (gkalpak) [reply aio/angular.json]: I am not sure, but I guess this was it: https://github.com/angular/angular/blob/4506230fbbae7e5c0e30b2b99096ed86751383d9/aio/.angular-cli.json#L69

Comment (filipesilva) [reply aio/angular.json]: Right you are, filed a bug: https://github.com/angular/angular-cli/issues/10173.

Meanwhile they can be added manually.

**Comment by gkalpak** [comment]
Closing in favor of #23234.

**Pull request closed without merging**